File: 2.sync_git.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

import subprocess
import sys
import os

# 如果是Windows系统，确保正确显示彩色文本
if os.name == 'nt':
    os.system('color')

def color_text(text, color_code):
    return f"\033[{color_code}m{text}\033[0m"

def run_git_command(command):
    try:
        result = subprocess.run(f"git {command}", shell=True, check=True, 
                              stdout=subprocess.PIPE, stderr=subprocess.PIPE,
                              universal_newlines=True)
        return result.stdout.strip()
    except subprocess.CalledProcessError as e:
        print(f"命令执行失败: {e}")
        print(f"错误输出: {e.stderr}")
        return None

print(color_text("开始拉取", 33))

# 先执行一次git pull并显示输出
subprocess.run("git pull", shell=True)

# 再次执行并获取结果
pull_result = run_git_command("pull")
if pull_result is not None:
    print(pull_result)

    if pull_result in ["Already up to date.", "已经是最新的。"]:
        print(color_text("拉取成功，开始上传", 33))
        
        run_git_command("add .")
        run_git_command("commit -m \"note\"")
        run_git_command("push")
        
        print("上传结束")
else:
    print(color_text("Git拉取失败，无法继续同步", 31))

print(color_text("按任意键退出", 33))

# 等待用户按键
if os.name == 'nt':  # Windows
    os.system('pause >nul')
else:  # Unix/Linux
    try:
        import termios
        import tty
        def getch():
            fd = sys.stdin.fileno()
            old_settings = termios.tcgetattr(fd)
            try:
                tty.setraw(sys.stdin.fileno())
                ch = sys.stdin.read(1)
            finally:
                termios.tcsetattr(fd, termios.TCSADRAIN, old_settings)
            return ch
        getch()
    except ImportError:
        input() 

File: 2.sync_git.js
#!/usr/bin/env node

const { execSync } = require('child_process');
const readline = require('readline');

// 显示彩色文本
function colorText(text, colorCode) {
  return `\x1b[${colorCode}m${text}\x1b[0m`;
}

console.log(colorText('开始拉取', 33));

try {
  // 先执行一次git pull
  execSync('git pull', { stdio: 'inherit' });
  
  // 再执行一次并获取结果
  const result = execSync('git pull', { encoding: 'utf8' }).trim();
  console.log(result);

  if (result === 'Already up to date.' || result === '已经是最新的。') {
    console.log(colorText('拉取成功，开始上传', 33));
    
    execSync('git add .', { stdio: 'inherit' });
    execSync('git commit -m "note"', { stdio: 'inherit' });
    execSync('git push', { stdio: 'inherit' });
    
    console.log('上传结束');
  }
} catch (error) {
  console.error(colorText('发生错误:', 31), error.message);
}

console.log(colorText('按任意键退出', 33));

// 等待用户按键后退出
readline.emitKeypressEvents(process.stdin);
if (process.stdin.isTTY) {
  process.stdin.setRawMode(true);
}

process.stdin.on('keypress', () => {
  process.exit();
}); 